tools/cldr-code/src/main/java/org/unicode/cldr/util/LocalePathValueListMatcher.java

```diff
@@ -0,0 +1,89 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class LocalePathValueListMatcher {
+
+    private static final Splitter SPLIT_SEMI_COLON = Splitter.on(';');
+
+    public static class LocalePathValueMatcher {
+        final Pattern localePattern;
+        final Pattern pathPattern;
+        final Pattern valuePattern;
+
+        public LocalePathValueMatcher(List<String> parts) {
+            localePattern =
+                    parts.size() < 1 || parts.get(0).isEmpty()
+                            ? null
+                            : Pattern.compile(parts.get(0));
+            pathPattern =
+                    parts.size() < 2 || parts.get(1).isEmpty()
+                            ? null
+                            : Pattern.compile(parts.get(1).replace("[@", "\\[@"));
```

Comment:
Ideally this should go in a more generic utility class. Or, alternately, `XPathParts x.toPatternString()` where it could handle any other escaping needed

Reply:
Agreed.